pkg/webhook/add_ingress.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2023 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"github.com/openkruise/kruise/pkg/webhook/ingress/validating"
+)
+
+func init() {
+	addHandlers(validating.HandlerMap)
```

Comment:
you'd better use `addHandlersWithGate`

Reply:
Do not add the handler if corresponding feature-gate is not enabled.

Reply:
@veophi Done, do the same change as `add_service.go`.